### PR TITLE
Enable lints in astar_search

### PR DESCRIPTION
**Pull request by nnmm**
clang-tidy also pointed out that the `fmod` function is not `constexpr`, so the function will never return a `constexpr` value.

### CODE REVIEW

File: planning/scenario_planning/parking/astar_search/src/astar_search.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 #include "astar_search/astar_search.hpp"
+
```
Comment (jilaada) [inline planning/scenario_planning/parking/astar_search/src/astar_search.cpp:16]: Curious, why was a new line added here?

**Review by jilaada — APPROVED** [review]
I was able to compile this packages and the tests ran without errors. Changes look good to me. Just a small question about the addition of a new line below:

File: planning/scenario_planning/parking/astar_search/src/astar_search.cpp (hunk 1 above)
Comment (nnmm) [reply planning/scenario_planning/parking/astar_search/src/astar_search.cpp:16]: As the "corresponding header file", I thought it should be in its own group. Cpplint thinks so too: If there were any C++ stdlib headers, they would be put after this header, but before the helper header by cpplint.

Comment (jilaada) [reply planning/scenario_planning/parking/astar_search/src/astar_search.cpp:16]: i see, thanks for the explanation!

**Pull request merged**